server/src/main/java/org/opensearch/plugins/PluginInfo.java

```diff
@@ -69,7 +71,7 @@ public class PluginInfo implements Writeable, ToXContentObject {
     private final String name;
     private final String description;
     private final String version;
-    private final Version opensearchVersion;
+    private final SemverRange opensearchVersionRange;
```

Comment:
Or do you think it would be better to follow the example DB suggested of `>=X.Y.Z`?

Reply:
Even in the case of >=X.Y.Z, it would be better to have support for multiple ranges to accommodate [>=X.Y.Z, <A.B.C] so that range can be bounded. I can change this to a list.